services/aws-s3.js

```diff
@@ -93,13 +93,13 @@ Slingshot.S3Storage = {
           directive.bucketUrl,
 
         download = _.extend(url.parse(directive.cdn || bucketUrl), {
-          pathname: payload.key
+          pathname: directive.bucket +"/" + payload.key
```

Comment:
Do not add bucket when using CDN.

`payload.key` needs to be appended to `directive.cdn` or `bucketUrl` as it was before, but without overriding any path that either `cdn` or `bucketUrl` may use. Also a `"/"` may need to be inserted in between.